libraries/AP_Networking/AP_Networking_Config.h

```diff
@@ -0,0 +1,84 @@
+#include <AP_HAL/AP_HAL_Boards.h>
+
+#ifndef AP_NETWORKING_ENABLED
+#define AP_NETWORKING_ENABLED 0
+#endif
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+    #define AP_NETWORKING_DHCP_AVAILABLE    LWIP_DHCP
+#else
+    #define AP_NETWORKING_DHCP_AVAILABLE    1 // for non-ChibiOS, assume it's available
+#endif
+
+
+// ---------------------------
+// Below are default params
+// ---------------------------
+
+#ifndef AP_NETWORKING_DEFAULT_OPTIONS
+    #define AP_NETWORKING_DEFAULT_OPTIONS 0
+#endif
+
+// Default DHCP
+#ifndef AP_NETWORKING_DEFAULT_DHCP_ENABLE
+#define AP_NETWORKING_DEFAULT_DHCP_ENABLE 0
```

Comment:
Have you considered turning DHCP on by default to match behavior of most OS's? An average user might struggle more with their OS-specific network changes than having a common way of adjusting params in ArduPilot that we can document.

Reply:
Thought about it.. but chances are pretty low that a DHCP server will be running on the same network as this. As a default it wouldn't hurt though

Reply:
Want to make a bet?

Reply:
Default is now DHCP Enabled